src/Task/Base/ParallelExec.php

```diff
@@ -172,6 +188,12 @@ public function run()
                         }
                     }
                     unset($running[$k]);
+                    if ($this->stopOnFail && $process->getExitCode() !== 0) {
+                        foreach ($running as $otherRunningProcess) {
+                            $otherRunningProcess->stop();
+                        }
+                        $queue = [];
```

Comment:
I don't touch `$running` array because it feel safer to me to let it handle by the main `while` cycle.